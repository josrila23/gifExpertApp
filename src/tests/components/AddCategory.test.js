import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe("Pruebas en <AddCategory />", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("Debe mostrarse varias veces", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola mundo";

    input.simulate("change", { target: { value } });

    expect(wrapper.find("p").text().trim()).toBe(value);
  });
  test("No debe postear la incformaciÃ³n con submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });
  test("Debe de llamar el setCategories y limipiar la caja de texto", () => {
    // 1. Simular el inputChange
    const value = "Hola mundo";
    const input = wrapper.find("input");
    input.simulate("change", { target: { value } });

    // 2. Simular el submit
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    // 3. setCategories se debe haber llamado
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    // 4. El valor del input debe estar ''
    const inputValue = wrapper.find("input").prop("value");
    expect(inputValue).toBe("");
  });
});
