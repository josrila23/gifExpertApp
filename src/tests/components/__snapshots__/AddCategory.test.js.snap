// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> Debe mostrarse varias veces 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
